### DISCUSSION TITLE
JWT Inspection.

### DISCUSSION THREAD

**Discussion by srelf-eschercloud** [gravitational/teleport#18116] [Q&A] [answered]
Im looking to find out exactly what a JWT token looks like so i can work out how to pass group/roles to a JWT supporting application. 

Any ides how to intercept a token so i can inspect it.

This is to enable passing role information down into grafana, so each user can be given a role in teleport that maps to a role in grafana via JWT.

**Comment by webvictim** [comment, marked as answer]
If you turn on the `debug_app: true` setting under `app_service`, this will expose a simple app that just dumps headers out to the browser window. You can access it using https://dumper.teleport.example.com (where `teleport.example.com` is your proxy domain)

The `Teleport-Jwt-Assertion` header is what you're looking for. If you paste that into something like https://jwt.io/ it'll give you the decoded version.

Here's an example of a decoded payload from my own cluster:
```json
{
  "aud": [
    "http://127.0.0.1:37491"
  ],
  "exp": 1667520826,
  "iat": 1667506428,
  "iss": "teleport.example.com",
  "nbf": 1667506418,
  "roles": [
    "access",
    "editor",
    "sysadmin",
    "allow-requests",
    "allow-user-root",
    "per-session-mfa",
    "allow-aws-readonly",
    "allow-mssql-access",
    "allow-mysql-access",
    "allow-postgres-access",
    "kubernetes-access-eks",
    "kubernetes-access-gke",
    "allow-request-approval",
    "windows-desktop-access",
    "allow-aws-ec2-fullaccess"
  ],
  "sub": "gus@example.com",
  "username": "gus@example.com"
}
```

**Reply by srelf-eschercloud** [reply]
Amazing thanks, once i could see what was being sent i was able to extract exactly what was needed

cheers.
